src/openrct2-ui/interface/Window.cpp

```diff
@@ -175,86 +175,58 @@ rct_window* window_create(
     return w;
 }
 
+static ScreenCoordsXY ClampWindowToScreen(const ScreenCoordsXY& pos, const int32_t screenWidth, const int32_t width)
+{
+    auto screenPos = pos;
+    if (screenPos.x < 0)
+        screenPos.x = 0;
+    if (screenPos.x + width > screenWidth)
+        screenPos.x = screenWidth - width;
```

Comment:
If we add similar clauses for the Y coordinate, this would resolve #12262.

Reply:
I can add it in a different commit or on a different PR, what do you prefer?

Reply:
Hmm, perhaps best to leave it for another PR, then. I don't want to cause this one to be scope-creeped.